### PR TITLE
fix: Snapshot Toggles Should Reset Correctly When Pinning New Command

### PR DESCRIPTION
**Pull request by davidmunechika**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1107
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1370

### User facing changelog
When a user pins a new command, the Highlight switch and Before/After toggle will be reset to initial states and stay in sync with what the Snapshot is showing.

### Additional details

Before:

https://user-images.githubusercontent.com/27690333/158886585-2e5d6a39-68b2-4440-a600-d88709172e96.mov

https://user-images.githubusercontent.com/27690333/158889748-c0d6b129-1db1-4dbe-9e9f-56cbbfa75f5c.mov

After:

https://user-images.githubusercontent.com/27690333/158888959-81247364-9650-4279-af29-bf321aeaacde.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [n/a] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/src/runner/SnapshotControls.vue [hunk 1]
```diff
@@ -17,6 +17,7 @@
         <SnapshotToggle
           v-if="shouldShowStateControls"
           :messages="snapshotMessages"
+          :active-index="snapshotStore.snapshot?.stateIndex ?? 0"
```
Comment (ZachJW34) [inline packages/app/src/runner/SnapshotControls.vue:20]: Don't need the `?? 0` if you have the `withDefaults`.

File: packages/app/cypress/e2e/runner/runner.ui.cy.ts [hunk 2]
```diff
@@ -94,6 +94,33 @@ describe('src/cypress/runner', () => {
       .should('have.class', 'command-is-pinned')
     })
 
+    it('correctly resets highlight toggle state when pinning new command', () => {
```
Comment (ZachJW34) [inline packages/app/cypress/e2e/runner/runner.ui.cy.ts:97]: You fixed the `Before | After` sync as well, can you write a test for that?

Comment (ZachJW34) [inline packages/app/cypress/e2e/runner/runner.ui.cy.ts:97]: You fixed the `Before | After` sync as well, can you write a test for that?

File: packages/app/src/runner/SnapshotHighlightControls.vue [hunk 3]
```diff
@@ -6,32 +6,26 @@
     >{{ t('runner.snapshot.toggleHighlights') }}</label>
     <Switch
       name="toggle-highlights"
-      :value="toggleHighlights"
+      :value="props.value"
       size="sm"
-      @update="toggleHighlights = !toggleHighlights"
+      @update="emits('update:value', !props.value)"
```
Comment (tbiethman) [inline packages/app/src/runner/SnapshotHighlightControls.vue:11]: It doesn't look like we ever read the boolean value we emit here, can we remove the value from the API?

**Comment by lmiller1990** [comment]
I reviewed the additional commits and tested this out - works great. I wonder if the "toggle highlights" label should also be clickable? Thoughts @marktnoonan ?

**Review by lmiller1990 — APPROVED** [review]

File: packages/app/cypress/e2e/runner/runner.ui.cy.ts (hunk 2 above)
Comment (lmiller1990) [reply packages/app/cypress/e2e/runner/runner.ui.cy.ts:97]: Tyler did here https://github.com/cypress-io/cypress/pull/20670/commits/074a97432504b8c3af0536c86c898e45f509c11a#diff-14b335c5128675493e52a73e003be1d2ae2013934caa565075a40ef4361fbcffR129

**Pull request merged**